### PR TITLE
[js/rn] Add test for validating "executionProvider" options

### PR DESCRIPTION
**Pull request by YUNQIUGUO**
### Description

As title.

Validation at JS call level in E2E app is not included. Can cover together in a separate pr.

### Motivation and Context

Test coverage.

### CODE REVIEW

File: js/react_native/android/src/androidTest/java/ai/onnxruntime/reactnative/OnnxruntimeModuleTest.java [hunk 1]
```diff
@@ -149,4 +156,49 @@ public void onnxruntime_module() throws Exception {
...
+      when(Arguments.createMap()).thenAnswer(i -> new JavaOnlyMap());
+      when(Arguments.createArray()).thenAnswer(i -> new JavaOnlyArray());
+
+      OnnxruntimeModule ortModule = new OnnxruntimeModule(reactContext);
+      ortModule.blobModule = blobModule;
+      String sessionKey = "";
+
+      // test loadModel() with nnapi ep options
+      {
+        InputStream modelStream =
```
Comment (edgchen1) [inline js/react_native/android/src/androidTest/java/ai/onnxruntime/reactnative/OnnxruntimeModuleTest.java:173]: do we need to close the InputStream? consider using [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html)

Comment (YUNQIUGUO) [reply js/react_native/android/src/androidTest/java/ai/onnxruntime/reactnative/OnnxruntimeModuleTest.java:175]: thanks for the suggestion and catch. added a modelstream.close() here for simplicity.

Comment (skottmckay) [reply js/react_native/android/src/androidTest/java/ai/onnxruntime/reactnative/OnnxruntimeModuleTest.java:175]: Would using `try` also ensure it's closed if there's an exception?

Comment (YUNQIUGUO) [reply js/react_native/android/src/androidTest/java/ai/onnxruntime/reactnative/OnnxruntimeModuleTest.java]: I see. updated to `try-with-resources` block which can ensure the resource will be closed in all cases.

**Review by skottmckay — APPROVED** [review]
:shipit:

**Pull request merged**